### PR TITLE
Notices: Add global notices tests

### PR DESCRIPTION
**Pull request by sirreal**
Add tests.
Small refactor changes to handle props and dismissal clearly.

Follow up to #24634
Likely would have prevented #24633

## Testing
1. Tests pass?
1. Are notices correctly dismissed? https://calypso.live/?branch=add/global-notices-tests&debug
   Clickable:
   ```js 
   dispatch( { type: 'NOTICE_CREATE', notice: {
   noticeId: 'testing',
   status: 'is-success',
   icon: 'thumbs-up',
   onDismissClick: () => console.log('Dismissed!'),
   showDismiss: true,
   text: 'Demo notice 💥',
   displayOnNextPage: true,
   isLoading: true,
   isCompact: true,
   } } ) 
   ```
   Auto-dismiss after 1s
   ```js 
   dispatch( { type: 'NOTICE_CREATE', notice: {
   noticeId: 'testing',
   status: 'is-success',
   text: 'Demo notice 💥',
   duration: 1000
   } } ) 
   ```

### CODE REVIEW

File: client/components/global-notices/index.jsx [hunk 1]
```diff
@@ -86,21 +89,25 @@ const NoticesList = createReactClass( {
...
-				);
-			}, this )
+			this.props.storeNotices.map(
+				// Collect `<Notice />` props in rest, extract other expected props
+				function( { button, href, noticeId, onClick, ...notice } ) {
+					return (
+						<Notice
+							{ ...notice }
+							key={ `notice-${ noticeId }` }
+							onDismissClick={ this.removeReduxNotice( noticeId, notice.onDismissClick ) }
```
Comment (sirreal) [inline client/components/global-notices/index.jsx]: Nit question: Should the `onDismissClick` be extracted as well?

**Comment by sirreal** [comment]
Rebased and added the necessary eslint ignores for the CI errors

**Comment by sirreal** [comment]
Rebased.

File: client/components/global-notices/index.jsx (hunk 1 above)
Comment (ockham) [reply client/components/global-notices/index.jsx]: Makes sense, I think. The less we rely on the order of props (our own `onDismissClick` overriding the one currently passed as part of `{ ...notice }`), the better.

File: client/components/global-notices/test/index.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+beforeEach( jest.clearAllMocks );
+
+describe( '<NoticesList />', () => {
+	test( 'should not render without notices', () => {
+		const wrapper = shallow( <NoticesList { ...baseProps } /> );
+		expect( wrapper.type() ).toBeNull();
+	} );
+
+	test( 'should render notices with expected HTML', () => {
```
Comment (ockham) [inline client/components/global-notices/test/index.js]: Super-picky, but maybe rephrase to `expected markup` or `expected DOM` or `expected JSX`? Simply because a `shallow` test doesn't really go as far as checking the actual HTML produced (as is reflected by the Jest snapshot).

**Comment by sirreal** [comment]
New lint rules around `componentWillMount` prompted some additional janitorial changes.
